modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/CustomReflectionObjectHandler.java

```diff
@@ -40,6 +44,14 @@ public Object coerce(Object object) {
 
         if (object.getClass().isArray()) {
             return new ArrayMap(object);
+        } else if (object instanceof Map) {
+            try {
+                @SuppressWarnings("unchecked")
+                JsonMap map = new JsonMap((Map<String, Object>) object);
+                return map;
+            } catch (ClassCastException e) {
```

Comment:
I don't think this can ever result in a `ClassCastException` because of type erasure.